onnx/version_converter/adapters/adapter.h

```diff
@@ -26,7 +26,7 @@ class Adapter {
         target_version_(target_version) {
     }
 
-    virtual void adapt(std::shared_ptr<Graph> /*graph*/, Node* node) const = 0;
+    virtual Node* adapt(std::shared_ptr<Graph> /*graph*/, Node* node) const = 0;
```

Comment:
does this create a new Node and returns it? Can you add comments for this mtd

Reply:
I had to make this change because in case of deprecated operators we have to destroy the current node and replace it with another one. Since the new node is referenced via its pointer even after `adapt` is called, `adapt` has to return the pointer. But 99% of the time (when the node is preserved and only its attributes/inputs/outputs are changed) the returned pointer is the same as the argument.

Reply:
you have graph and mutable Node* can you not create a new node and point Node* to this new node?

Reply:
Yes but why bother to create a new node, copy all the attributes/inputs/outputs of the old node to the new one and destroying the old node in all adapters when we can almost always modify the old node in place? That's how the adapters have always worked up to just before this PR. I have added the possibility of returning a different node only to deal with deprecated operators.